DSharpPlus/Entities/Interaction/Application/DiscordApplicationCommand.WeakEquals.cs

```diff
@@ -0,0 +1,282 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DSharpPlus.Entities;
+
+partial class DiscordApplicationCommand
+{
+    /// <summary>
+    /// Implements a weak-equality check for application commands, treating null and default values the same
+    /// and only comparing fields known locally.
+    /// </summary>
+    public bool WeakEquals(DiscordApplicationCommand other)
+    {
+        return this.Name == other.Name
+            && this.Description == other.Description
+            && (this.DefaultMemberPermissions ?? DiscordPermissions.None) 
+                == (other.DefaultMemberPermissions ?? DiscordPermissions.None)
+            && (this.NSFW ?? false) == (other.NSFW ?? false)
+            && this.Type == other.Type
+            && IntegrationTypesMatch(this.IntegrationTypes, other.IntegrationTypes)
+            && EnumListsMatch(this.Contexts, other.Contexts)
+            && LocalizationsMatch(this.NameLocalizations, other.NameLocalizations)
+            && LocalizationsMatch(this.DescriptionLocalizations, other.DescriptionLocalizations)
+            && OptionsMatch(this.Options, other.Options);
+    }
+
+    private static bool LocalizationsMatch
+    (
+        IReadOnlyDictionary<string, string> a,
+        IReadOnlyDictionary<string, string> b
+    )
+    {
+        // if both are null or empty, they are equivalent
+        if ((a is null || a.Count == 0) && (b is null || b.Count == 0))
+        {
+            return true;
+        }
+
+        // if one of the two is null, but not both (as per above), they are not equivalent
+        if (a is null || b is null)
+        {
+            return false;
+        }
+
+        // they are both not-null, so let's go check
+        // if they aren't evenly long, they by necessity cannot be equivalent
+        if (a.Count != b.Count)
+        {
+            return false;
+        }
+
+        foreach (KeyValuePair<string, string> val in a)
+        {
+            if (!b.TryGetValue(val.Key, out string? remoteValue) || val.Value != remoteValue)
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static bool OptionsMatch
+    (
+        IReadOnlyList<DiscordApplicationCommandOption>? a,
+        IReadOnlyList<DiscordApplicationCommandOption>? b
+    )
+    {
+        // if both are null or empty, they are equivalent
+        if ((a is null || a.Count == 0) && (b is null || b.Count == 0))
+        {
+            return true;
+        }
+
+        // if one of the two is null, but not both (as per above), they are not equivalent
+        if (a is null || b is null)
+        {
+            return false;
+        }
+
+        // they are both not-null, so let's go check
+        // if they aren't evenly long, they by necessity cannot be equivalent
+        if (a.Count != b.Count)
+        {
+            return false;
+        }
```

Comment:
Nit: Why aren't we using `else if`'s? These all return

Reply:
why would we use `else if`'s, there's no functional difference